plugin/core/rpc.py

```diff
@@ -333,26 +282,31 @@ def receive_payload(self, message: str) -> None:
             except Exception as err:
                 exception_log("Error handling {}".format(typestr), err)
 
-    def on_transport_closed(self) -> None:
-        self._error_display_handler("Communication to server closed, exiting")
-        # Differentiate between normal exit and server crash?
-        if not self.exiting:
-            self.handle_transport_failure()
+    def on_stderr_message(self, message: str) -> None:
+        pass
+
+    def on_transport_close(self, exit_code: int, exception: Optional[Exception]) -> None:
+        self.transport = None
 
     def response_handler(self, response_id: int, response: Dict[str, Any]) -> Tuple[Optional[Callable], Any]:
         handler, error_handler = self._response_handlers.pop(response_id, (None, None))
+        if not handler:
```

Comment:
Is everything still working correctly with sync requests after this change? Those don't add anything to `_response_handlers`, I believe.

Reply:
Ah! So that's why the formatting tests failed in test_single_document :)

Reply:
Nicely spotted